cpp/test/utils/metrics/log_metric_test.cc

```diff
@@ -0,0 +1,150 @@
+                          metric_dimension_names);
+  ASSERT_TRUE(GetMetricValuesFromLogs().empty());
+
+  test_metric.AddOrUpdate(metric_dimension_values, 1.0);
+  test_metric.AddOrUpdate(metric_dimension_values, -2.0);
+  test_metric.AddOrUpdate(metric_dimension_values, metric_request_id, -2.5);
+  test_metric.AddOrUpdate(metric_dimension_values, metric_request_id, 3.5);
+  const std::vector<double> expected_metric_values{1.0, 3.5};
+  ASSERT_EQ(GetMetricValuesFromLogs(), expected_metric_values);
+}
+
+TEST_F(TSLogMetricTest, TestGaugeMetric) {
+  TSLogMetric test_metric(MetricType::GAUGE, metric_name, "count",
+                          metric_dimension_names);
+  ASSERT_TRUE(GetMetricValuesFromLogs().empty());
+
+  test_metric.AddOrUpdate(metric_dimension_values, 1.0);
+  test_metric.AddOrUpdate(metric_dimension_values, -2.0);
+  test_metric.AddOrUpdate(metric_dimension_values, metric_request_id, 3.5);
+  test_metric.AddOrUpdate(metric_dimension_values, metric_request_id, -4.0);
+  const std::vector<double> expected_metric_values{1.0, -2.0, 3.5, -4.0};
+  ASSERT_EQ(GetMetricValuesFromLogs(), expected_metric_values);
+}
+
+TEST_F(TSLogMetricTest, TestHistogramMetric) {
+  TSLogMetric test_metric(MetricType::HISTOGRAM, metric_name, "count",
+                          metric_dimension_names);
+  ASSERT_TRUE(GetMetricValuesFromLogs().empty());
+
+  test_metric.AddOrUpdate(metric_dimension_values, 1.0);
+  test_metric.AddOrUpdate(metric_dimension_values, -2.0);
+  test_metric.AddOrUpdate(metric_dimension_values, metric_request_id, 3.5);
+  test_metric.AddOrUpdate(metric_dimension_values, metric_request_id, -4.0);
+  const std::vector<double> expected_metric_values{1.0, -2.0, 3.5, -4.0};
+  ASSERT_EQ(GetMetricValuesFromLogs(), expected_metric_values);
+}
+
+TEST_F(TSLogMetricTest, TestTSLogMetricEmitWithRequestId) {
+  TSLogMetric test_metric(MetricType::COUNTER, metric_name, "ms",
+                          metric_dimension_names);
+  test_metric.AddOrUpdate(metric_dimension_values, metric_request_id, 1.5);
+  const std::vector<std::string> metric_logs = GetMetricLogs();
+  const std::string expected_metric_log_pattern =
+      "^.*\\[METRICS\\]test_metric\\.Milliseconds\\:1\\.5\\|#level\\:model,"
+      "model_name\\:test_model\\"
+      "|#hostname\\:.*,[0-9]+,test_request_id?$";
+  ASSERT_EQ(metric_logs.size(), 1);
+  ASSERT_THAT(metric_logs[0],
+              ::testing::MatchesRegex(expected_metric_log_pattern));
+}
+
+TEST_F(TSLogMetricTest, TestTSLogMetricEmitWithoutRequestId) {
+  TSLogMetric test_metric(MetricType::COUNTER, metric_name, "ms",
+                          metric_dimension_names);
+  test_metric.AddOrUpdate(metric_dimension_values, 1.5);
+  const std::vector<std::string> metric_logs = GetMetricLogs();
+  const std::string expected_metric_log_pattern =
+      "^.*\\[METRICS\\]test_metric\\.Milliseconds\\:1\\.5\\|#level\\:model,"
+      "model_name\\:test_model\\"
+      "|#hostname\\:.*,[0-9]+$";
```

Comment:
[This](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/metrics/Metric.java#L15) is how frontend parse backend metrics log. Please follow the same regex.

Reply:
Good pint, will update unit tests to use same metric log regex as frontend.